src/components/sort_button/SortButton.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useState } from "react";
+import ArrowIcon from "../../assets/arrow-icon.png";
+import { HeaderContainer, SortIcon } from "../sort_button/SortButton.styles";
+
+interface SortableHeaderProps {
+  text: string;
+  onSort?: (asc: boolean) => void;
+}
+
+export const SortableButton: React.FC<SortableHeaderProps> = ({
+  text,
+  onSort,
+}) => {
+  const [asc, setAsc] = useState(true);
+
+  const handleClick = () => {
+    setAsc(!asc);
+    onSort?.(!asc);
+  };
+
+  return (
+    <HeaderContainer onClick={handleClick}>
+      {text}
+      <SortIcon src={ArrowIcon} alt="sort" $rotated={!asc} />
+    </HeaderContainer>
```

Comment:
Sobre esto un par de comentarios:

Si bien HTML te permite poner texto renderizado directamente en un div como `<div>Texto</div>` esto no es una buena práctica ya que los textos deberían estar renderizados directamente en los elementos HTML para esto como `<p>`, `<h1>`, etc

Por eso deberiamos hacer algo como:


```suggestion
    <HeaderContainer onClick={handleClick}>
      <Text>{text}</Text>
      <SortIcon src={ArrowIcon} alt="sort" $rotated={!asc} />
    </HeaderContainer>
```

Donde `<Text>` sea un elemento del tipo `<p>`.
Ahora, otra cosa es que estuve viendo que los titulos de las columnas no tienen un tamaño de texto definido, deberiamos agregarlo y que el tamaño cambie segun el tamaño de la pantalla